crates/cheatnet/src/forking/state.rs

```diff
@@ -45,6 +52,38 @@ impl ForkStateReader {
     }
 }
 
+impl BlockInfoReader for ForkStateReader {
+    fn get_block_info(&mut self) -> StateResult<CheatnetBlockInfo> {
+        if let Some(cache_hit) = self.cache.get_block_info() {
+            return Ok(cache_hit);
+        }
+
+        match self
+            .runtime
+            .block_on(self.client.get_block_with_tx_hashes(self.block_id))
+        {
+            Ok(MaybePendingBlockWithTxHashes::Block(block)) => {
+                let block_info = CheatnetBlockInfo {
+                    block_number: BlockNumber(block.block_number),
+                    timestamp: BlockTimestamp(block.timestamp),
+                    sequencer_address: ContractAddress(patricia_key!(block.sequencer_address)),
+                };
+
+                self.cache.cache_get_block_info(block_info);
+
+                Ok(block_info)
+            }
+            Ok(MaybePendingBlockWithTxHashes::PendingBlock(_)) => Err(StateReadError(format!(
+                "Block with id {:?} is pending",
+                self.block_id
+            ))),
```

Comment:
I am not sure if this is correct semantics, we should investigate it

Reply:
Wdym not correct, the match?

Reply:
I mean returning the error.

Reply:
changed it, now `pending_block.number = pending_block.parent_block.number + 1`

Reply:
Although I didn't add a test for forking a pending block, because it fails to fetch it sometimes: `StarknetError(StarknetErrorWithMessage { code: Known(BlockNotFound), message: \"Block not found\" })`

Probably at certain moments, there is no `Pending` block

Reply:
Same with `Latest`, I think it's fine to include those

Reply:
I'm afraid such a test will make the CI fail too often

Reply:
If that actually happens more often than once in a while (the one with `Latest` happened once for me in CI, locally a few times) then it's fine to not include it

Reply:
after an internal discussion, we have decided that it would be better to remove `pending` block forking https://github.com/foundry-rs/starknet-foundry/issues/938